pkg/dispers/dispers/requests.go

```diff
@@ -27,3 +32,178 @@ type InputCI struct {
 type OutputCI struct {
 	Hashes []Concept `json:"hashes,omitempty"`
 }
+
+/*
+*
+Target Finders' Input & Output
+*
+*/
+
+func union(a, b []string) []string {
+	m := make(map[string]bool)
+
+	for _, item := range a {
+		m[item] = true
+	}
+
+	for _, item := range b {
+		if _, ok := m[item]; !ok {
+			a = append(a, item)
+		}
+	}
+	return a
+}
+
+func intersection(a, b []string) (c []string) {
+	m := make(map[string]bool)
+
+	for _, item := range a {
+		m[item] = true
+	}
+
+	for _, item := range b {
+		if _, ok := m[item]; ok {
+			c = append(c, item)
+		}
+	}
+	return
+}
+
+// NodeType are the only possible nodes in Target Profile trees
+type NodeType int
+
+const (
+	// SingleNode are Target Profile's leafs
+	SingleNode NodeType = iota
+	// UnionNode are unions between two lists
+	UnionNode
+	// IntersectionNode are intersections between two lists
+	IntersectionNode
+)
```

Comment:
I am not familiar with this syntax ; the `iota` means all the elements in the const will have an incremental integer value, starting from 0?
And why is `NodeType ` specified on the first line but not the others? Is it a syntax shortcut?

Reply:
I don't pretend to be familiar either. I found this on Stackoverflow. It s quite like a enum in Java/Python/Whatever. I simply copied and that the way they does. 

SingleNode takes the value 0
UnionNode 1
IntersectionNode 2  

The three of them are NodeType !